compiler/src/middle_end/linearize.re

```diff
@@ -63,16 +63,42 @@ let lookup_symbol = (mod_, mod_decl, name, original_name) => {
   };
 };
 
-type anf_bind =
-  | BSeq(comp_expression)
-  | BLet(Ident.t, comp_expression)
-  | BLetRec(list((Ident.t, comp_expression)))
-  | BLetExport(rec_flag, list((Ident.t, comp_expression)));
-
 type either('a, 'b) =
   | Left('a)
   | Right('b);
 
+let convert_binds = anf_binds => {
+  let void_comp = Comp.imm(Imm.const(Const_void));
+  let void = AExp.comp(void_comp);
+  let (last_bind, top_binds) =
+    switch (anf_binds) {
+    | [bind, ...rest] => (bind, rest)
+    | _ => (BSeq(void_comp), [])
```

Comment:
Is there any need to match on `[bind]` and use an empty list for the top_binds?

Reply:
The `[bind, ...rest]` case should handle that. 🙂

Reply:
Ha, well I'll be. For some reason my brain thought that only matched lists with 2+ elements.